ops/services/alerts/app_service_metrics/exceptions.tf

```diff
@@ -0,0 +1,53 @@
+# To do log-based alerts, we need to make sure we're actually collecting AppServiceConsoleLogs from the API
+data "azurerm_resource_group" "management" {
+  name = "prime-simple-report-management"
+}
+
+data "azurerm_log_analytics_workspace" "global" {
+  name                = "simple-report-log-workspace-global"
+  resource_group_name = data.azurerm_resource_group.management.name
+}
+
+resource "azurerm_monitor_diagnostic_setting" "collect_appserviceconsolelogs" {
+  name                       = "API App Service console logs"
+  target_resource_id         = var.app_service_id
+  log_analytics_workspace_id = data.azurerm_log_analytics_workspace.global.id
+
+  log {
+    category = "AppServiceConsoleLogs"
+    enabled  = true
+
+    retention_policy {
+      days    = 7
+      enabled = true
+    }
+  }
+}
+
+# Add an alert for GraphQL query validation failures (more than 3 in a 5-minute window)
+resource "azurerm_monitor_scheduled_query_rules_alert" "graphql_query_validation_failures" {
+  name                = "graphql-query-validation-failures"
+  location            = "East US"
+  resource_group_name = "prime-simple-report-${var.env}"
+
+  action {
+    action_group = [var.action_group_id]
+  }
+
+  data_source_id = var.app_service_id
+  description    = "Alert when GraphQL query validation failures occur in the API"
+  enabled        = true
+
+  query = <<-QUERY
+AppServiceConsoleLogs
+  | where tolower(ResultDescription) contains "Query failed to validate"
+  QUERY
+
+  severity    = 1
+  frequency   = 5
+  time_window = 5
+  trigger {
+    operator  = "GreaterThan"
+    threshold = 3
```

Comment:
This seems like something that should be configurable, or at least should be called out as a named constant (I feel like those are equivalent, since we can have a variable with a default value).

Reply:
I agree, but I wonder if there's a way to do a larger refactor of this later. I don't know if this scales with a lot of alerts, since if we do it here we should do it for the other 3 alerts too. It seems like we'd need a map variable per alert to contain the config. Thoughts?

Reply:
Having looked at it a little, I think the threshold here should actually be 0, and not be configurable—if we start getting too many alerts, we can worry about it then.

Reply:
Seems reasonable to me - just changed it to 0